packages/react-components/react-dialog/src/components/Dialog/useDialog.ts

```diff
@@ -54,9 +58,20 @@ export const useDialog_unstable = (props: DialogProps): DialogState => {
     requestOpenChange,
   });
 
-  const handlebackdropClick = useEventCallback((event: React.MouseEvent<HTMLDivElement>) => {
+  const { targetDocument } = useFluent_unstable();
+  const disableScroll = useDisableScroll();
+  const isBodyScrollLocked =
+    useDialogContext_unstable(ctx => ctx.isBodyScrollLocked) || Boolean(open && modalType !== 'non-modal');
```

Comment:
Can you please clarify what does `useDialogContext_unstable(ctx => ctx.isBodyScrollLocked)`?

Reply:
It's for the rare case of nested Dialogs, otherwise, closing the internal Dialog would already enable the body scroll meanwhile it should be disabled, it's exactly the same previous idea of having a counter of how many times disable scroll has been called to enable it after, but in a more indirect way I'd say.

Reply:
There is one benefit in counters, actually 🐱 Here is an example:

```tsx
function App1() {
  return (
    <Dialog>
      <Dialog />
    </Dialog>
  );
}
```

```tsx
function App2() {
  return (
    <>
      <Dialog />
      <Dialog />
    </>
  );
}
```

The approach with context will work for `App1`, but it will not work for `App2` as they are not in the same part of a tree. I don't know our limitations for dialog nesting, so it might not be a problem.

Reply:
That is definitely a strange (almost impossible) use case, but yeah... it could happen, you got a point. I'll rollback to the counter version.